src/data-workspace/section-form/section.js

```diff
@@ -40,8 +41,17 @@ export const SectionFormSection = ({
             (grp) => grp.categoryCombo.categoryOptionCombos?.length || 1
         )
     )
-    const filterInputId = `filter-input-${section.id}`
 
+    const greyedFieldsMap = section.greyedFields.reduce((acc, greyedField) => {
+        const key = getFieldId(
+            greyedField.dataElement,
+            greyedField.categoryOptionCombo
+        )
+        acc[key] = true
+        return acc
+    }, {})
```

Comment:
I think this could be simplified by using a `Set` object:
```js
const greyedFields = new Set(section.greyedFields.map(greyedField => getFieldId(greyedField.dataElement, greyedField.categoryOptionCombo)))
// ...
const isGreyedField = greyedFields.has(fieldId)
```

Reply:
Yeah, I just tend to be cautious with Map/Sets when sharing them as props, since they can easily cause update/render bugs. But that is mostly when using `setState` I guess, since here we are creating a new object each render anyways.